### PR TITLE
Support of filters

### PR DESCRIPTION
**Pull request by andpopov**
Part of: #1254

### CODE REVIEW

**Comment by andpopov** [comment]
@olenagerasimova, @dgarus Please review

File: src/main/java/com/artipie/settings/cache/CachedFilters.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Implementation of cache for filters using {@link LoadingCache}.
+ *
+ * @since 0.28
+ * @checkstyle DesignForExtensionCheck (500 lines)
+ */
+public class CachedFilters implements FiltersCache {
+    /**
+     * Cache for storages.
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/settings/cache/CachedFilters.java:27]: java doc seems off

File: src/main/java/com/artipie/settings/cache/CachedFilters.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public String toString() {
+        return String.format(
+            "%s(size=%d)",
+            this.getClass().getSimpleName(), this.cache.size()
+        );
+    }
+
+    @Override
+    public void invalidate(final YamlMapping repoyaml) {
+        this.cache.invalidate(repoyaml);
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/settings/cache/CachedFilters.java:72]: hm, key of the cache is `String` reponame, why are we invalidating by some yaml?

Comment (olenagerasimova) [reply src/main/java/com/artipie/settings/cache/CachedFilters.java:72]: Also, invalidation method should be called from `RepositoryRest` to clean the cache when repo settings were changed... Corresponding test correction is also required.

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@andpopov thanks, please, check my comments

File: src/main/java/com/artipie/settings/cache/CachedFilters.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     * Cache for storages.
+     */
+    private final Cache<String, Optional<Filters>> cache;
+
+    /**
+     * Ctor.
+     */
+    public CachedFilters() {
+        this.cache = CacheBuilder.newBuilder()
+            .expireAfterWrite(
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/settings/cache/CachedFilters.java:36]: hm, why `expireAfterWrite`? Would not `expireAfterRead` be more suitable here?

**Comment by andpopov** [comment]
@olenagerasimova Thanks for importan comments, please take a look at changes.

**Review by olenagerasimova — DISMISSED** [review]
@andpopov thanks, looks good to me

File: src/main/java/com/artipie/settings/cache/CachedFilters.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of cache for filters using {@link LoadingCache}.
+ *
+ * @since 0.28
+ * @checkstyle DesignForExtensionCheck (500 lines)
+ */
+public class CachedFilters implements FiltersCache {
```
Comment (dgarus) [inline src/main/java/com/artipie/settings/cache/CachedFilters.java:25]: Suggest to rename `CachedFilters` to `GuavaFiltersCache`.

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks!
There is a minor comment.

File: src/main/java/com/artipie/settings/cache/CachedFilters.java (hunk 4 above)
Comment (andpopov) [reply src/main/java/com/artipie/settings/cache/CachedFilters.java]: @dgarus Good idea!
Renamed, please review

**Review by dgarus — APPROVED** [review]

**Pull request merged**